Update profiler features status for .NET 10 (#32001)
DataDog/documentation: content/en/profiler/profile_types.md

Thrown Exceptions (v2.31+)
: The number of caught or uncaught exceptions raised by each method, as well as their type and message.

Allocations (v3.28+)
: The number and size of allocated objects by each method, as well as their type.
For .NET Framework, the size is not available.<br />

> _Requires: .NET Framework (with Datadog Agent 7.51+ and v3.2+) / .NET 6+ but .NET 10+ is recommended for more accurate sampling_

Reviewer: ```suggestion
_Requires: .NET Framework (with Datadog Agent 7.51+ and v3.2+) / .NET 6+, but Datadog recommends .NET 10+ for more accurate sampling.
```

Lock (v2.49+)
: The number of times threads are waiting for a lock and for how long.<br />
_Requires: .NET Framework (requires Datadog Agent 7.51+) / .NET 5+_

Live Heap (v3.28+)
: A subset of the allocated objects (with their class name) that are still in memory.<br />

> _Requires: .NET 7+ but .NET 10+ is recommended for more accurate sampling_

Reviewer: ```suggestion
_Requires: .NET 7+ but Datadog recommends .NET 10+ for more accurate sampling.
```

: The time garbage collector's threads spent running on the CPU.<br />
_Requires: .NET Framework (with Datadog Agent 7.51+ and v3.2+) / .NET 5+_

> Note: Before .NET 10, **Allocations** and **Live Heap** profiling might show bigger objects more than smaller ones due to sampling algorithm used by the .NET runtime. It is recommended to use .NET 10+ for more statistically correct results. 

Reviewer: ```suggestion
**Note**: Before .NET 10, **Allocations** and **Live Heap** profiling might show bigger objects more than smaller ones due to the sampling algorithm used by the .NET runtime. Datadog recommends using .NET 10+ for more statistically correct results. 
```
